### PR TITLE
Add --fork option

### PR DESCRIPTION
**Pull request by sileht**
This options fallbacks to no fork when repository have disabled forking
    by default.

    Behavior can be forced by adding "always", "never", "auto".

### CODE REVIEW

**Comment by sileht** [comment]
I have hesitated to make it the default behavior, instead of adding a new option.

File: git_pull_request/__init__.py [hunk 1]
```diff
@@ -386,16 +386,25 @@ def preserve_older_revision(branch, remote_to_push):
...
-        head = "{}:{}".format(repo_to_fork.owner.login, branch)
-    else:
-        repo_forked = g_user.create_fork(repo_to_fork)
+    forked = False
+    if fork:
+        try:
+            repo_forked = g_user.create_fork(repo_to_fork)
+        except github.GithubException as e:
+            if (fork == "auto" and e.status == 403 and
+                    "forking is disabled" in e.data["message"]):
```
Comment (jd) [inline git_pull_request/__init__.py]: Can you add

```python
LOG.debug("Forking is disabled on target repository")
```

**Comment by sileht** [comment]
> Maybe we could use `--fork <value>` where value is true/false/auto with default `auto`?
Cool, my initial version was doing, then I tryied to not break old option.

**Comment by sileht** [comment]
> I'm envisioning that we'll also support settings the default via `git config` which would make `git config git-pull-request.fork` returns the behavior that has been configured.

**Pull request merged**